convert-project.py

```diff
@@ -1,65 +1,143 @@
 import csv
-import os
 import json
 import sys
-from typing import Generator
+from typing import Generator, Any
+from pathlib import Path
 
 
-def check_file_exist(file_path: str) -> bool:
+def check(file_path: str) -> tuple[bool, Path]:
     """
-    入力したファイルが自分のPCに入っているのかを確認する関数
-    :param file_path: jsonのデータ形式に変換したいファイル
-    :return: ファイルが存在していればTrue、存在していなかったらFalseを返す
+    入力したファイルのパスを確認する関数
+    :param file_path: 入力したファイルのパス
+    :return: 入力したファイルのパス・csvならTrue,jsonならFalseを返す
     """
-    if not os.path.exists(file_path):
-        print('Error : Input file does not exist.')
-        return False
-    return True
+    if not isinstance(file_path, str):
+        raise TypeError(f"Argument 'file-path' must be a string, but received type {type(file_path)}.")
+    input_file_path: Path = Path(file_path)
+    if not input_file_path.exists():
+        raise FileNotFoundError(f"File path '{input_file_path}' does not exist.")
+    if not input_file_path.is_file():
+        raise ValueError(f"File path '{input_file_path}' is not a file.")
+    if input_file_path.suffix not in [".csv", ".json"]:
+        raise ValueError(f"File path '{input_file_path}' does not have extension '.csv' or '.json'.")
+    if input_file_path.suffix.lower() == '.csv':
+        is_csv = True
+    elif input_file_path.suffix.lower() == '.json':
+        is_csv = False
+    else:
+        raise ValueError(f"File path '{input_file_path}' does not have extension '.csv' or '.json'.")
```

Comment:
- 文字列はダブルクォートで統一しましょう。
- 21行目で チェック済なので、このelse分岐は不要だと思います。

私ならこう書きます。
```suggestion
    is_csv = True if input_file_path.suffix.lower() == ".csv" else False
    # または
    is_csv = input_file_path.suffix.lower() == ".csv"
```

Reply:
承知しました。
28行目の部分を削除し、「is_csv = True if input_file_path.suffix.lower() == ".csv" else False」に変更しました。